src/material/datepicker/datepicker.md

```diff
@@ -480,6 +480,9 @@ information about `MAT_DATE_FORMATS`.
 export class MyApp {}
 ```
 
+If you need to work with native `Date` objects, but need custom behavior (for example custom date
+parsing), you can consider subclassing `NativeDateAdapter`.
```

Comment:
I think developers might be confused about what to pass for the `matDateLocale` in the constructor.

```
/** Adapts the native JS Date for use with cdk-based components that work with dates. */
@Injectable()
export class NativeDateAdapter extends DateAdapter<Date> {
  constructor(
    @Optional() @Inject(MAT_DATE_LOCALE) matDateLocale: string,
    /**
     * @deprecated No longer being used. To be removed.
     * @breaking-change 14.0.0
     */
    _platform?: Platform,
  ) { ... }
  ...
}
```

There are multiple ways to set the locale as documented in [ Setting the locale code
](https://material.angular.io/components/datepicker/overview#setting-the-locale-code.). I would suggest injecting MAT_DATE_LOCALE into the class that's calling the constructor, so that it can be passed along. I'm not sure we want to go as far as creating an example for extending NativeDateAdapter.

Reply:
I'd prefer to actually just move it out of the constructor and use `inject()`. I'll add a commit to do that